packages/nextjs/components/scaffold-stark/FaucetSepolia.tsx

```diff
@@ -0,0 +1,139 @@
+import Image from "next/image";
+
+/**
+ * Faucet modal which lets you send ETH to any address.
+ */
+export const FaucetSepolia = () => {
+  const { chain: ConnectedChain } = useNetwork();
+  const { targetNetwork } = useTargetNetwork();
+
+  const sepoliaFaucets = [
+    {
+      name: "Starknet Foundation",
+      img: "/sn-symbol-gradient.png",
+      link: "https://starknet-faucet.vercel.app/",
+    },
+    {
+      name: "Alchemy",
+      img: "/logo_alchemy.png",
+      link: "https://www.alchemy.com/faucets/starknet-sepolia",
+    },
+    {
+      name: "Blast",
+      img: "/blast-icon-color.svg",
+      link: "https://blastapi.io/faucets/starknet-sepolia-eth",
+    },
+  ];
+
+  const publicNodeUrl = targetNetwork.rpcUrls.public.http[0];
+
+  // Use useMemo to memoize the publicClient object
+  const publicClient = useMemo(() => {
+    return new RpcProvider({
+      nodeUrl: publicNodeUrl,
+    });
+  }, [publicNodeUrl]);
+
+  useEffect(() => {
+    const checkChain = async () => {
+      try {
+        const providerInfo = await publicClient.getBlock();
+        console.log(providerInfo);
+      } catch (error) {
+        console.error("⚡️ ~ file: Faucet.tsx:checkChain ~ error", error);
+        notification.error(
+          <>
+            <p className="font-bold mt-0 mb-1">
+              Cannot connect to local provider
+            </p>
+            <p className="m-0">
+              - Did you forget to run{" "}
+              <code className="italic bg-base-300 text-base font-bold">
+                yarn chain
+              </code>{" "}
+              ?
+            </p>
+            <p className="mt-1 break-normal">
+              - Or you can change{" "}
+              <code className="italic bg-base-300 text-base font-bold">
+                targetNetwork
+              </code>{" "}
+              in{" "}
+              <code className="italic bg-base-300 text-base font-bold">
+                scaffold.config.ts
+              </code>
+            </p>
+          </>,
+          {
+            duration: 5000,
+          }
+        );
+      }
+    };
+    checkChain().then();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  // Render only on sepolia chain
+  if (ConnectedChain?.id !== sepolia.id) {
+    return null;
+  }
+
+  return (
+    <div>
+      <label
+        htmlFor="faucet-modal"
+        className="btn btn-sm font-normal gap-1 border border-[#32BAC4] shadow-none"
+      >
+        <BanknotesIcon className="h-4 w-4 text-[#32BAC4]" />
+        <span>Faucet</span>
+      </label>
+      <input type="checkbox" id="faucet-modal" className="modal-toggle" />
+      <label htmlFor="faucet-modal" className="modal cursor-pointer">
+        <label className="modal-box relative">
+          {/* dummy input to capture event onclick on modal box */}
+          <input className="h-0 w-0 absolute top-0 left-0" />
+          <h3 className="text-xl font-bold mb-6">Sepolia Faucets</h3>
```

Comment:
we can add a small text saying :

Disclaimer:
Please note that these external websites are provided for your convenience. We do not have control over the content and availability of these sites. Use at your own risk.

Reply:
Noted